core-bundle/src/Resources/contao/library/Contao/Files.php

```diff
@@ -156,6 +156,14 @@ public function fopen($strFile, $strMode)
 	{
 		$this->validate($strFile);
 
+		$filesystem = System::getContainer()->get('filesystem');
+		$parentDir = \dirname($this->strRootDir . '/' . $strFile);
+
+		if (!$filesystem->exists($parentDir))
+		{
+			$filesystem->mkdir($parentDir);
+		}
+
```

Comment:
No need to check for existence or ask the container IMO.

```suggestion
        (new Filesystem())->mkdir(\dirname($this->strRootDir . '/' . $strFile));
```

Reply:
but that will make it impossible to mock …

Reply:
Mocking is not needed here, see the `fopen()` in the next line.

Reply:
Changed in 0e37183f2689cd33c2c6b7df34e02c57b197423e